### PR TITLE
[WIP] - ECS Integration

### PR DESCRIPTION
**Pull request by kochie**
This PR adds support for Amazon ECS as the backend container service.

The functionality will allow users to

- [x] Use Fargate/Fargate Spot to run containers.
- [ ] Define Task Definition files for different shiny apps.
- [ ] Integration and Unit tests.

### CODE REVIEW

**Comment by LEDfan** [comment]
Hi @kochie 

Thanks for working on this, looks very interesting! I was wondering whether you need any help? As you may have noticed in the last release, some major changes were made in the backend API. This should make it a lot easier to implement new backends, but this also means that the current code may need some changes.

**Comment by kochie** [comment]
Hey @LEDfan thanks for the offer! Yes it would be great to have some input. I've actually completed this for a client already and we we're able to get it running perfectly in their account. There were some specific issues related to their environment that can be fixed up, happy to discuss in code review. I'll update this branch with the latest version.

File: src/main/java/eu/openanalytics/containerproxy/backend/ecs/EcsBackend.java [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+    public List<ExistingContainerInfo> scanExistingContainers() throws Exception {
+        // TODO Auto-generated method stub
+
+        ArrayList<ExistingContainerInfo> containers = new ArrayList<>();
+
+        ListTasksResponse ecsTasks = ecsClient.listTasks(builder -> builder.cluster(getProperty(PROPERTY_CLUSTER)));
+        List<String> taskArns = ecsTasks.taskArns();
+
+        DescribeTasksResponse tasks = ecsClient
+                .describeTasks(builder -> builder.cluster(getProperty(PROPERTY_CLUSTER)).tasks(taskArns));
```
Comment (christiandt) [inline src/main/java/eu/openanalytics/containerproxy/backend/ecs/EcsBackend.java:107]: Really unsure if I am reading this one correctly, but will this implementation handle the case where `taskArns` contains more than 100 items? https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/ecs/model/DescribeTasksRequest.Builder.html

**Comment by LEDfan** [comment]
Hi, we released ShinyProxy 3.1.0 today, which contains full support for ECS. It's based on this contribution, but was extended and fully integration in ShinyProxy. Thank you again for your contribution!

**Pull request merged**